src/components/popUp/popUpSrv.js

```diff
@@ -0,0 +1,156 @@
+'use strict';
+
+(function () {
+    angular.module('znk.infra.popUp').factory('PopUpSrv',[
+        '$injector', '$q', '$rootScope', '$animate',
+        function ($injector, $q, $rootScope, $animate) {
+            var PopUpSrv = {};
+
+            var $body = angular.element(document.querySelector('body'));
+            $body.append('<div class="znk-popup"></div>');
+            var popUpsPlaceHolderElement = angular.element(document.querySelector('.znk-popup'));
+
+            var popupInstance,
+                popupDefer;
+
+            PopUpSrv.closePopup = function(reject,reason){
+                if(!reason){
+                    reason = 'closed';
+                }
+                popUpsPlaceHolderElement.empty();
+                if (popupInstance.scope) {
+                    popupInstance.scope.$destroy();
+                }
+                popupDefer[(reject ? 'reject' : 'resolve')](reason);
+            };
+
+            PopUpSrv.popup = function popup(wrapperCls,header,body,buttonsArr){
+                //kill current popup if exists
+                if(popupInstance){
+                    PopUpSrv.closePopup();
+                }
+                var childScope = $rootScope.$new(true);
+                childScope.d = {};
+
+                popupDefer = $q.defer();
+                popupInstance = {};
+                popupInstance.promise = popupDefer.promise;
+
+                var template =
+                    '<div class="%wrapperCls%">' +
+                    '<div class="znk-popup-wrapper">' +
+                    '<div class="znk-popup-header">%header%</div>' +
+                    '<div class="znk-popup-body">%body%</div>' +
+                    '<div class="znk-popup-buttons">' +
+                    '<div ng-repeat="button in d.buttons" class="button-wrapper">' +
+                    '<button class="btn" ng-click="d.btnClick(button)" ng-class="button.type" analytics-on="click" analytics-event="click-popup-{{button.text}}" analytics-category="popup">{{button.text}}</button>' +
+                    '</div>' +
+                    '</div>' +
+                    '</div>';
+
+                wrapperCls = wrapperCls ? ' ' + wrapperCls : '';
+                template = template.replace('%wrapperCls%',wrapperCls);
+
+                header = header || '';
+                template = template.replace('%header%',header);
+
+                body = body || '';
```

Comment:
What are the benefits ? 
I prefer it this way since it not polluting the global closure.